File: mailGonderme/app/src/main/java/com/example/mailgonderme/MainActivity.java
package com.example.mailgonderme;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText adres,konu,icerik;
    Button gonder;
    String icerikT,adresT,konuT;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tanimla();
        gonder();
    }
    public void tanimla(){
        adres=(EditText) findViewById(R.id.adres);
        konu=(EditText)findViewById(R.id.konu);
        icerik=(EditText)findViewById(R.id.icerik);
        gonder=(Button)findViewById(R.id.gonder);
    }
    public void bilgiAl(){
        icerikT=icerik.getText().toString();
        adresT=adres.getText().toString();
        konuT=konu.getText().toString();
    }
    public void gonder(){
        gonder.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bilgiAl();
                Intent intent=new Intent(Intent.ACTION_SEND);
                intent.setType("message/rfc822");
                intent.putExtra(Intent.EXTRA_EMAIL,adresT);
                intent.putExtra(Intent.EXTRA_SUBJECT,konuT);
                intent.putExtra(Intent.EXTRA_TEXT,icerikT);
                startActivity(intent);
            }
        });
    }
}

File: KotlinDers1StoringData/app/src/main/java/com/example/kotlinders1storingdata/MainActivity.kt
package com.example.kotlinders1storingdata

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    //var sharedPreferences: SharedPreferences?=null
    //ya böyle tanımlıycaz ya da
    lateinit var sharedPreferences: SharedPreferences
    var ageFromPreferences: Int? = null
    var age: Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //sharedPreferences
        //name: veritabanı adımızdır genelde açtığımız package name'i veririz.
        //mode: oluşturulan dosya sadece kendi uygulaması içinde ulaşılsın istediğimiz için modumuz private
        sharedPreferences = this.getSharedPreferences(
            "com.example.kotlinders1storingdata",
            Context.MODE_PRIVATE
        )
        ageFromPreferences = sharedPreferences.getInt("age", -1)
        if (ageFromPreferences == -1) {
            textView.text = "Your Age: "
        } else {
            textView.text = "Your Age: $ageFromPreferences"
        }
        age = 71
    }

    fun delete(view: View) {
        ageFromPreferences = sharedPreferences.getInt("age", -1)
        if (ageFromPreferences != -1) {
            //kayıtlı veri var
            //edit kullanıyorsak apply da dememiz gerekiyor.
            sharedPreferences.edit().remove("age").apply()
            textView.text = "Your Age: "

        }
    }

    fun save(view: View) {

        val myAge = ageText.text.toString().toIntOrNull()
        if (myAge != null) {
            textView.text = "Your Age: $myAge"
            //ya da
            //textView.text="Your Age: "+myAge
            sharedPreferences.edit().putInt("age", myAge).apply()
        }
    }

    fun next(view: View) {

        val intent = Intent(applicationContext, Main2Activity::class.java)
        intent.putExtra("age", age)
        startActivity(intent)
    }
}
